# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1291937801 18000
#      Thu Dec 09 18:36:41 2010 -0500
# Node ID 703f51c2beb72c524bdd51ee9738f158d014a14b
# Parent  492e7b587352fee1cb7adfb73fc5a1d1f88bab29
set grid linewidths from axis linewidth property

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-12-09  John W. Eaton  <jwe@octave.org>
+
+	* plot/__go_draw_axes__.m: Set major and grid linewidths from
+	axis linewidth property.
+
 2010-12-09  Rik  <octave@nomad.inbox5.com>
 
 	* plot/__print_parse_opts__.m: Convert print resolution from char to
 	double.  Fixes Bug #31834.
 
 2010-12-09  Rik  <octave@nomad.inbox5.com>
 
 	* plot/private/__pie__.m: Allow logical indices for 'explode' variable.
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -319,16 +319,20 @@ function __go_draw_axes__ (h, plot_strea
       fputs (plot_stream, "set border front;\n");
     else
       fputs (plot_stream, "set grid layerdefault;\n");
       ## FIXME -- the gnuplot help says that "layerdefault" should work
       ## for set border too, but it fails for me with gnuplot 4.2.5.  So
       ## use "back" instead.
       fputs (plot_stream, "set border back;\n");
     endif
+
+    fprintf (plot_stream, "set grid linewidth %f, linewidth %f;\n",
+             axis_obj.linewidth, axis_obj.linewidth);
+
     if (! have_grid)
       fputs (plot_stream, "unset grid;\n");
     endif
 
     do_tics (axis_obj, plot_stream, ymirror, mono, gnuplot_term);
 
     fputs (plot_stream, "unset logscale;\n");
     xlogscale = strcmpi (axis_obj.xscale, "log");
